api/src/services/notifications/groups/notify-admins-of-added-user-service.ts

```diff
@@ -0,0 +1,53 @@
+import { Op } from "@sequelize/core"
+
+import { Group, User, Notification } from "@/models"
+
+import BaseService from "@/services/base-service"
+import { Notifications } from "@/services"
+
+export class NotifyAdminsOfAddedUserService extends BaseService {
+  constructor(
+    private user: User,
+    private group: Group,
+    private currentUser: User
+  ) {
+    super()
+  }
+
+  async perform() {
+    await this.group.reload({ include: ["adminUsers"] })
+
+    const groupAdmins = this.group.adminUsers ?? []
+
+    const groupAdminIds = groupAdmins.map((user) => user.id)
+
+    const systemAdmins = await User.findAll({
+      where: {
+        roles: {
+          [Op.like]: `%${User.Roles.SYSTEM_ADMIN}%`,
+        } as { [Op.like]: string },
+        id: {
+          [Op.notIn]: groupAdminIds,
+        },
+      },
+    })
+
+    const { name: groupName } = this.group
+    const safeTitle = Notification.sanitizeAttribute("title", `User Added to Group: ${groupName}`)
+
+    for (const user of [...systemAdmins, ...groupAdmins]) {
```

Comment:
You seem to be doing this query a lot, might be worth adding a scope like users who are admins of a group or are system admins?